x/compute/internal/types/errors.go

```diff
@@ -2,39 +2,59 @@ package types
-	ErrAccountExists = sdkErrors.Register(DefaultCodespace, 2, "contract account already exists")
+	ErrCreateFailed = sdkErrors.Register(DefaultCodespace, 1, "create contract failed")
 
 	// ErrInstantiateFailed error for rust instantiate contract failure
-	ErrInstantiateFailed = sdkErrors.Register(DefaultCodespace, 3, "instantiate wasm contract failed")
+	ErrInstantiateFailed = sdkErrors.Register(DefaultCodespace, 2, "instantiate contract failed")
 
 	// ErrExecuteFailed error for rust execution contract failure
-	ErrExecuteFailed = sdkErrors.Register(DefaultCodespace, 4, "execute wasm contract failed")
+	ErrExecuteFailed = sdkErrors.Register(DefaultCodespace, 3, "execute contract failed")
+
+	// ErrQueryFailed error for rust smart query contract failure
+	ErrQueryFailed = sdkErrors.Register(DefaultCodespace, 4, "query contract failed")
+
+	// ErrMigrationFailed error for rust execution contract failure
+	ErrMigrationFailed = sdkErrors.Register(DefaultCodespace, 5, "migrate contract failed")
+
+	// ErrAccountExists error for a contract account that already exists
+	ErrAccountExists = sdkErrors.Register(DefaultCodespace, 6, "contract account already exists")
 
 	// ErrGasLimit error for out of gas
-	ErrGasLimit = sdkErrors.Register(DefaultCodespace, 5, "insufficient gas")
+	ErrGasLimit = sdkErrors.Register(DefaultCodespace, 7, "insufficient gas")
 
 	// ErrInvalidGenesis error for invalid genesis file syntax
-	ErrInvalidGenesis = sdkErrors.Register(DefaultCodespace, 6, "invalid genesis")
+	ErrInvalidGenesis = sdkErrors.Register(DefaultCodespace, 8, "invalid genesis")
 
 	// ErrNotFound error for an entry not found in the store
-	ErrNotFound = sdkErrors.Register(DefaultCodespace, 7, "not found")
-
-	// ErrQueryFailed error for rust smart query contract failure
-	ErrQueryFailed = sdkErrors.Register(DefaultCodespace, 8, "query wasm contract failed")
+	ErrNotFound = sdkErrors.Register(DefaultCodespace, 9, "not found")
 
 	// ErrInvalidMsg error when we cannot process the error returned from the contract
-	ErrInvalidMsg = sdkErrors.Register(DefaultCodespace, 9, "invalid CosmosMsg from the contract")
-
-	// ErrMigrationFailed error for rust execution contract failure
-	ErrMigrationFailed = sdkErrors.Register(DefaultCodespace, 10, "migrate wasm contract failed")
+	ErrInvalidMsg = sdkErrors.Register(DefaultCodespace, 10, "invalid CosmosMsg from the contract")
 )
+
+func IsEncryptedErrorCode(code uint32) bool {
+	return 5 >= code && code > 0
+}
+
+func ErrContainsQueryError(err error) bool {
+	return strings.Contains(err.Error(), ErrQueryFailed.Error())
+}
+
+// ** Warning **
+// Below are functions that check for magic strings that depends on the output of the enclave.
+// Beware when changing this, or the rust error string
+func ContainsEnclaveError(str string) bool {
+	return strings.Contains(str, "Enclave")
+}
+
+func ContainsEncryptedString(str string) bool {
+	return strings.Contains(str, "encrypted: ")
+}
```

Comment:
I would change this to use something like "StartsWith" if possible. If the contract is able to inject plaintext information to the strings that get checked by these functions, then they can mess with this logic.

Reply:
It's not startswith because these won't be at the start of the string. Also, all the inputs that users can create is encrypted afaik, so it's hard to make this act up. Finally, it only affects queries, so I would be more concerned about a mistake causing something weird to be displayed rather than malicious behaviour.

Reply:
i see... if you think that it's safe and sound, then we can resolve this conversation.